consensus/miner_signalled_consensus.md

```diff
@@ -0,0 +1,247 @@
+# Miner Signalled Consensus Upgrade
+
+A consensus protocol upgrade based on miner signalling outcome can be referred
+to as a _conditional_ consesus upgrade. A hard-fork (in the context of a
+consensus protocol upgrade based on miner signalling) happens with a key block,
+therefore, using the term "block" in the following text refers to a "key
+block". A consensus protocol upgrade not based on any condition is
+_unconditional_ protocol upgrade.
+
+## Simple Summary
+
+This document describes how nodes in the network signal the readiness for a new
+consensus protocol, its activation, and a hard-fork at a certain height.
+
+## Motivation
+
+Activating a new consensus protocol based on miner signalling outcome prevents
+the block chain to diverge in case a large fraction of the miners have not
+upgraded their nodes to a version able to follow the new consensus protocol; or
+in case the majority of the miners is against consensus protocol upgrade.
+
+## Specification
+
+### Overview
+
+On a node configured for a consensus upgrade based on miner signalling outcome,
+the new consensus protocol becomes active at a determined height if at least a
+determined majority of the blocks between two determined block heights
+(signalling interval) contains a predefined signal. Such consesus protocol
+upgrade happens per chain fork.
+
+The signalling interval has a determined number of confirmations before the
+determined height at which the new consensus protocol may become active (fork
+height). This is in order to:
+- Prevent the node from having to aggregate the results of the signalling
+  interval in a too short time interval.
+- Prevent a user from restarting the node and configuring the new consensus
+  protocol unconditionally when the signalling outcome is not confirmed with
+  reasonably high probability.
+
+The signal value is part of a block header in the signalling interval:
+- Is meant to be a pseudorandom number, so to limit the risk of the same value
+  being used by simultaneous competing consensus protocol proposals.
```

Comment:
What if someone chooses the same value intentionally?

Reply:
When the majority of the mining power signals a new protocol and then also follows it, then the nodes that were signalling the new protocol, but now are not able to follow it, will end up on a forked chain.